drivers/dma/dma_mcux_lpc.c

```diff
@@ -191,12 +202,75 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+		LOG_ERR("DMA attempted to configure an Otrig mux"
+		" while the mux was in use.");
+		return -EACCES;
+	}
+
+	if (config->source_chaining_en || config->dest_chaining_en) {
+		/* Chaining is enabled */
+		LOG_DBG("link dma 0 channel %d with channel %d",
+			channel, config->linked_channel);
+		uint8_t is_otrig_available = 0;
+
+		for (otrig_index = 0; otrig_index < dev_config->num_of_otrigs;
+			++otrig_index) {
+			if (dma_data->otrig_array[otrig_index].linked_channel == -1 ||
+			    dma_data->otrig_array[otrig_index].source_channel == channel) {
+				if (dma_data->otrig_array[otrig_index].source_channel == channel) {
+					DMA_DisableChannel(DEV_BASE(dev),
+						dma_data->otrig_array[otrig_index].linked_channel);
+					dma_data->num_channels_used--;
```

Comment:
You cannot subtract from this variable- `num_channels_used` tracks the number of virtual channels the driver has configured. Since the driver does not offer a way to "unconfigure" a virtual channel, we cannot decrease the number of allocated channels here.

Reply:
These have been removed.